### PR TITLE
Don't run VCR tests if there is a build failure

### PR DESCRIPTION
**Pull request by trodge**
fixes https://github.com/hashicorp/terraform-provider-google/issues/14657

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/scripts/go-plus/gcb-terraform-vcr-tester/test_terraform_vcr.sh [hunk 1]
```diff
@@ -97,7 +97,11 @@ export GOOGLE_TEST_DIRECTORY=$(go list ./... | grep -v github.com/hashicorp/terr
 echo "checking terraform version"
 terraform version
 
-
+go build $GOOGLE_TEST_DIRECTORY
+if [ $? != 0 ]; then
+  echo "Skipping tests: Build failure detected"
```
Comment (ScottSuarez) [inline .ci/scripts/go-plus/gcb-terraform-vcr-tester/test_terraform_vcr.sh:102]: VCR status will remain pending at this point due to line 55. Could we post the pending status after we run confirm build-ability?

**Review by ScottSuarez — COMMENTED** [review]
How did the benchmarking go? Were there any reasonable impact? 

Even if so it would be the same if we were to chain this off of another step that builds the provider (ie unit tests).

**Comment by trodge** [comment]
> How did the benchmarking go? Were there any reasonable impact?
> 
> Even if so it would be the same if we were to chain this off of another step that builds the provider (ie unit tests).

I wasn't able to measure any impact because the build takes 1-2 minutes and the total time varies by more than that.

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**